includes/CLI/Plugin_List_Check_Categories_Command.php

```diff
@@ -0,0 +1,179 @@
+	 */
+	public function __construct( Plugin_Context $plugin_context ) {
+		$this->plugin_context = $plugin_context;
+	}
+
+	/**
+	 * Runs plugin list-check-categories.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--fields=<fields>]
+	 * : Limit displayed results to a subset of fields provided.
+	 *
+	 * [--format=<format>]
+	 * : Format to display the results. Options are table, csv, and json. The default will be a table.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *   wp plugin list-check-categories
+	 *   wp plugin list-check-categories --format=json
+	 *
+	 * @subcommand list-check-categories
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $args       List of the positional arguments.
+	 * @param array $assoc_args List of the associative arguments.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 */
+	public function list_check_categories( $args, $assoc_args ) {
+		// Get options based on the CLI arguments.
+		$options = $this->get_options( $assoc_args );
+
+		// Get categories details.
+		$categories = $this->get_categories();
+
+		// Get formatter.
+		$formatter = $this->get_formatter( $options );
+
+		// Display results.
+		$formatter->display_items( $categories );
+	}
+
+	/**
+	 * Returns check categories details.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of the check categories.
+	 */
+	private function get_categories() {
+		$categories = array();
+
+		$check_categories = new Check_Categories();
+		$categories_slugs = $check_categories->get_categories();
+
+		if ( is_array( $categories_slugs ) && count( $categories_slugs ) > 0 ) {
+			$categories = array_map(
+				function ( $slug ) {
+					return array(
+						'slug' => $slug,
+						'name' => ucfirst( str_replace( '_', ' ', $slug ) ),
+					);
+				},
+				$categories_slugs
+			);
+
+		}
+
+		return $categories;
+	}
+
+	/**
+	 * Validates the associative arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args List of the associative arguments.
+	 * @return array List of the associative arguments.
+	 *
+	 * @throws WP_CLI\ExitException Show error if plugin not found.
+	 */
+	private function get_options( $assoc_args ) {
+		$defaults = array(
+			'format' => 'table',
+		);
+
+		$options = wp_parse_args( $assoc_args, $defaults );
+
+		if ( ! in_array( $options['format'], $this->output_formats, true ) ) {
+			WP_CLI::error(
+				sprintf(
+					// translators: 1. Output formats.
+					__( 'Invalid format argument, valid value will be one of [%1$s]', 'plugin-check' ),
+					implode( ', ', $this->output_formats )
+				)
+			);
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Gets the formatter instance to format check results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args Associative arguments.
+	 * @return WP_CLI\Formatter The formatter instance.
+	 */
+	private function get_formatter( $assoc_args ) {
```

Comment:
By combining the classes, this method could be removed since `Plugin_Check_Command` already has it. The only difference in the three commands is the `$default_fields`, which we can easily solve by making it a new parameter on the private method. That way all 3 command methods can use this one single method and pass their own needed `$default_fields`.